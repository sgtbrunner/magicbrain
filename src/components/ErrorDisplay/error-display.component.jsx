import React from 'react';
import PropTypes from 'prop-types';

const ErrorDisplay = ({ error }) => (
  <p className="red f6 mv3 pa3 right-0 left-0 b bg-light-pink ba b--red br2" role="alert">
    {error}
  </p>
);

ErrorDisplay.propTypes = {
  error: PropTypes.string.isRequired,
};

export default ErrorDisplay;
