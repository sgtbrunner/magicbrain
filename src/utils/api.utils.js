const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
