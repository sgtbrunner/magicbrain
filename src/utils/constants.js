export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
