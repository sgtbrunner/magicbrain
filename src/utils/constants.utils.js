const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
