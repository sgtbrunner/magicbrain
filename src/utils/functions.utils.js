export const getUserFromLocalStorage = () => localStorage.getItem('user');
