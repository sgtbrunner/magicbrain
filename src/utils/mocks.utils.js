export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
