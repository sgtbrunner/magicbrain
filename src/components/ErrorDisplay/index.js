import ErrorDisplay from './error-display.component';

export default ErrorDisplay;
