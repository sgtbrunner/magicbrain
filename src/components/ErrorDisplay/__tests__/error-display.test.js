import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import ErrorDisplay from '..';

const ERROR = 'Oooops...an error occurred';

const mountErrorDisplayComponent = () =>
  mountComponent({ component: ErrorDisplay, props: { error: ERROR } });

describe('ErrorDisplay component', () => {
  it('should match snapshot', () => {
    const { container } = mountErrorDisplayComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render component with correct error message', () => {
    mountErrorDisplayComponent();

    const error = screen.getByRole('alert');
    expect(error).toHaveTextContent(ERROR);
  });
});
