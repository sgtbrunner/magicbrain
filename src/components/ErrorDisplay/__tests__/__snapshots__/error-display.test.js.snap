// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay component should match snapshot 1`] = `
<div>
  <p
    class="red f6 mv3 pa3 right-0 left-0 b bg-light-pink ba b--red br2"
  >
    Oooops...an error occurred
  </p>
</div>
`;
